### PR TITLE
[READY] Timestop rework from TG

### PR DESCRIPTION
**Pull request by Couls**
**What does this PR do:**
Where do I start? new effects better tracking better sounds more things frozen like thrown objects and even mechs!

**Images of sprite/map changes (IF APPLICABLE):**
ezgif-5-0f32c99932e8

**Changelog:**
:cl:
add: new fields subsystem from TG
add: new timestop effects
tweak: thrown objects now freeze mid-air from timestops
tweak: mechs now get frozen as well
tweak: new negative effect for everything caught in timestops
tweak: timestop zones are better handled
tweak: new timestop sound for when it ends
fix:timestop immune mobs no longer get stuck in the circle
fix: can no longer shoot guns and disarm while frozen in time
/:cl:

### CODE REVIEW

**Comment by taukausanake** [comment]
steamuserimages-a akamaihd
Za warudo intensifies

File: code/modules/reagents/chemistry/recipes/slime_extracts.dm [hunk 1]
```diff
@@ -612,7 +612,6 @@
 	var/obj/effect/timestop/T = new /obj/effect/timestop
 	T.forceMove(get_turf(holder.my_atom))
 	T.immune += mob
-	T.timestop()
```
Comment (taukausanake) [inline code/modules/reagents/chemistry/recipes/slime_extracts.dm]: Is this removing timestop from slime reactions? Won't that make sepia slimes meaningless now?

Comment (Couls) [reply code/modules/reagents/chemistry/recipes/slime_extracts.dm]: No timestop is automatically started when the effect is made now

File: code/modules/fields/timestop.dm [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+	for(var/A in immune_atoms)
+		immune[A] = TRUE
+	for(var/mob/living/L in GLOB.player_list)
+		if(locate(/obj/effect/proc_holder/spell/aoe_turf/conjure/timestop) in L.mind.spell_list) //People who can stop time are immune to its effects
+			immune[L] = TRUE
+	for(var/mob/living/simple_animal/hostile/guardian/G in GLOB.mob_list)
+		if(G.summoner && locate(/obj/effect/proc_holder/spell/aoe_turf/conjure/timestop) in G.summoner.mind.spell_list) //It would only make sense that a person's stand would also be immune.
+			immune[G] = TRUE
+	if(start)
+		timestop()
```
Comment (Couls) [inline code/modules/fields/timestop.dm]: @taukausanake

**Comment by Evankhell561** [comment]
this... 
https://github.com/tgstation/tgstation/pull/47695

**Comment by Couls** [comment]
Out of scope, I’m not adding to this outside of fixing issues after it’s reviewed

**Pull request closed without merging**